app/controllers/api/v1/events_controller.rb

```diff
@@ -1,5 +1,38 @@
-class Api::V1::EventsController < ApplicationController
+class Api::V1::EventsController < ApiController
   def show
     @event = Event.find(params[:id])
   end
+
+  def create
+    @event = Event.new(event_params)
+    if @event.save
+      render
+    else
+      render json: {
+        message: 'Validation Failed',
+        errors: @event.errors.full_messages
+      }, status: 422
+    end
+  end
+
+  private
+
+  def event_params
+    params.require(:event).permit(
+      :address,
+      :ended_at,
+      :lat,
+      :lon,
+      :name,
+      :started_at).
+      merge(owner: user)
```

Comment:
I think leaving this as-is would be fine, but another way to do this would be to make another method similar to how @brimatteng did here: https://github.com/brimatteng/bt-humon/blob/master/app/controllers/api/v1/events_controller.rb#L39